tools/slang-cpp-extractor/slang-cpp-extractor-diagnostic-defs.h

```diff
@@ -0,0 +1,45 @@
+//
+
+// The file is meant to be included multiple times, to produce different
```

Comment:
Okay, I think I get it. You are re-using the existing diagnostics infrastructure for the extractor, which means that it is helpful to have the diagnostics system not have too many dependencies on the rest of the compiler (which relies on the output of the extractor to build).

It seems like we need to figure out a holistic policy for this kind of stuff where the Slang compiler proper needs a service that some of our utility/testing/build code may also need, but the version for the Slang compiler may benefit from a degree of customization for and integration with the specifics of the compiler. Right now there are three main strategies we've got going on:

1. Have two copies of some service (e.g., the way that there is a tokenizer/lexer in `core`, and then the real one in `slang`)
2. Move the Slang compiler functionality out to `core`, even in cases where it isn't a very "universal" piece of code (e.g., the downstream compiler abstraction), and bring along any pieces of Slang-specific infrastructure/types that it needs
3. This case, where a piece of the Slang compiler proper is set up to be usable without the rest of the compiler in place, so that another tool/system can use it.

I worry that none of those solutions are really ideal.